### DISCUSSION TITLE
Does scikit-build-core plan to support Cython compilation?

### DISCUSSION THREAD

**Discussion by vyasr** [scikit-build/scikit-build-core#1457] [Q&A] [answered]
AFAICT `Python_add_library` is, roughly speaking, scikit-build-core's answer to scikit-build's `python_extension_module`. In the case of Cython files with scikit-build it was possible to use `add_cython_target` to translate Cython files into C(++) files  prior to creating the extension module. Is that something that scikit-build-core will support as well?

**Comment by henryiii** [comment, marked as answer]
Yes, that's planned. Probably a rewrite designed to be pretty close to the original, unless there's a reason to deviate. But the original now supports `FindPython`, so it's possible we could just include it.

**Reply by henryiii** [reply]
FYI, there's an example of using scikit-build-core with Cython (without a helper module yet) in the docs now.

**Reply by vyasr** [reply]
Are you referring to the [Cython tab here](https://scikit-build-core.readthedocs.io/en/latest/getting_started.html#cmake-file)?

**Reply by vyasr** [reply]
FYI I have opened discussions to add this support directly to either [CMake](https://gitlab.kitware.com/cmake/cmake/-/issues/24982) or [Cython](https://github.com/cython/cython/issues/5486). I'll follow up here based on how those conversations go, but feel free to weigh in if you have thoughts @henryiii or others.

**Reply by henryiii** [reply]
cython-cmake is the recommended way to go for now. It might be a good base for a Cython upstream contribution once it's used a bit more.
